librz/core/ctypes.c

```diff
@@ -178,269 +165,317 @@ RZ_IPI void rz_types_enum_print_c_all(RzTypeDB *typedb, bool multiline) {
 	rz_list_free(enumlist);
 }
 
-// Structured types (structures and unions)
-
-static bool sdb_if_union_cb(void *p, const char *k, const char *v) {
-	return !strncmp(v, "union", strlen("union") + 1);
-}
+// Unions
 
-static bool sdb_if_struct_cb(void *user, const char *k, const char *v) {
-	rz_return_val_if_fail(user, false);
-	Sdb *TDB = (Sdb *)user;
-	if (!strcmp(v, "struct") && !rz_str_startswith(k, "typedef")) {
-		return true;
-	}
-	if (!strcmp(v, "typedef")) {
-		const char *typedef_key = sdb_fmt("typedef.%s", k);
-		const char *type = sdb_const_get(TDB, typedef_key, NULL);
-		if (type && rz_str_startswith(type, "struct")) {
-			return true;
+static void core_types_union_print(RzCore *core, RzBaseType *btype, RzOutputMode mode, PJ *pj) {
+	rz_return_if_fail(core && btype);
```

Comment:
Probably good to also `return_if_fail` for `btype->kind == RZ_BASE_TYPE_KIND_UNION`

Reply:
Good idea. Done.